utilities/requestParser.js

```diff
@@ -1,34 +1,70 @@
 const _ = require('lodash');
+const moment = require('moment');
+const momentRange = require('moment-range');
 const queryString = require('query-string');
 
 const isParamValid = (inputParam, paramValidators) =>
-  _.every(_.values(paramValidators), validate => validate(inputParam));
+  _.every(_.values(paramValidators), (validate) => validate(inputParam));
 
 const validateInputParams = (inputParams, inputFormat) =>
   _.reduce(
     inputParams,
     (isValid, userInputProperty, propertyKey) =>
       isParamValid(userInputProperty, inputFormat[propertyKey]) && isValid,
-    true,
+    true
   );
 
+const getDatesBetween = (minDate, maxDate) => {
+  const min = moment(minDate, 'YYYY/MM/DD');
+
+  const max = moment(maxDate, 'YYYY/MM/DD');
+
+  if (
+    !min.isValid() &&
+    !max.isValid() &&
+    !min.isSameOrAfter(moment(), 'date') &&
+    !max.isSameOrAfter(moment(), 'date')
+  ) {
+    return [];
+  }
+  return _.map(
+    min.isSameOrBefore(max)
+      ? [ ...momentRange.extendMoment(moment).range(min, max).reverseBy('days') ]
```

Comment:
No sure what it does. I propose to extend that to separate method (or even `dateService`). And create method like

```
const getDatesInRange = () => {...}
```